CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativePreferenceManager.cs

```diff
@@ -58,12 +59,30 @@ internal long GetLong(string key, long defaultValue)
             return long.Parse(PlayerPrefs.GetString(GetStorageKey(key), defaultValue.ToString()));
         }
 
-        internal void SetLong(string key, long value)
+        internal void SetLong(string key, string longValue)
         {
-            // Use string to set long values in PlayerPrefs
-            PlayerPrefs.SetString(GetStorageKey(key), value.ToString());
+            PlayerPrefs.SetString(GetStorageKey(key), longValue);
         }
 
+        public void SetDouble(string key, string doubleValue)
+        {
+            PlayerPrefs.SetString(GetStorageKey(key), doubleValue.ToString());
+        }
+        
+        public double GetDouble(string key, double defaultValue)
+        {
+            return double.Parse(PlayerPrefs.GetString(GetStorageKey(key), defaultValue.ToString(CultureInfo.InvariantCulture)));
+        }
+        public void SetBool(string key, int value)
```

Comment:
Set bool should use a bool value as parameter.

Add new line before method.

Reply:
Done